### ISSUE TITLE
[5.0] Calling release on a Beanstalkd job that was buried throws an exception

### ISSUE THREAD

**Issue by hskrasek** [laravel/framework#9044]
I am trying to use Queued Commands/Jobs to handle delayed actions in an application. We are using Beanstalkd, and if an error happens we would like to bury the job so we can investigate and go from there.

What I have noticed is that if I bury the job and not throw an exception, the job is automatically deleted all together. I decided to solve this by burying the job and then re-throwing an exception I am catching to log errors and bury the job. This works, except for the fact the following exception is thrown:

```
exception 'Pheanstalk\Exception\ServerException' with message 'Job 13 NOT_FOUND: does not exist or is not reserved by client' in /home/vagrant/project/vendor/pda/pheanstalk/src/Command/ReleaseCommand.php:63
````

Now this doesn't break anything, but would like to remove the exception from happening altogether if possible. 

Full stack trace:
```
exception 'Pheanstalk\Exception\ServerException' with message 'Job 13 NOT_FOUND: does not exist or is not reserved by client' in /home/vagrant/project/vendor/pda/pheanstalk/src/Command/ReleaseCommand.php:63
Stack trace:
#0 /home/vagrant/project/vendor/pda/pheanstalk/src/Connection.php(121): Pheanstalk\Command\ReleaseCommand->parseResponse('NOT_FOUND', NULL)
#1 /home/vagrant/project/vendor/pda/pheanstalk/src/Pheanstalk.php(384): Pheanstalk\Connection->dispatchCommand(Object(Pheanstalk\Command\ReleaseCommand))
#2 /home/vagrant/project/vendor/pda/pheanstalk/src/Pheanstalk.php(259): Pheanstalk\Pheanstalk->_dispatch(Object(Pheanstalk\Command\ReleaseCommand))
#3 /home/vagrant/project/vendor/laravel/framework/src/Illuminate/Queue/Jobs/BeanstalkdJob.php(88): Pheanstalk\Pheanstalk->release(Object(Pheanstalk\Job), 1024, '0')
#4 /home/vagrant/project/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(217): Illuminate\Queue\Jobs\BeanstalkdJob->release('0')
#5 /home/vagrant/project/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(159): Illuminate\Queue\Worker->process('beanstalkd', Object(Illuminate\Queue\Jobs\BeanstalkdJob), '0', '0')
#6 /home/vagrant/project/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(107): Illuminate\Queue\Worker->pop(NULL, 'release-points', '0', '3', '0')
#7 /home/vagrant/project/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(67): Illuminate\Queue\Console\WorkCommand->runWorker(NULL, 'release-points', '0', '128', false)
#8 [internal function]: Illuminate\Queue\Console\WorkCommand->fire()
#9 /home/vagrant/project/vendor/laravel/framework/src/Illuminate/Container/Container.php(523): call_user_func_array(Array, Array)
#10 /home/vagrant/project/vendor/laravel/framework/src/Illuminate/Console/Command.php(115): Illuminate\Container\Container->call(Array)
#11 /home/vagrant/project/vendor/symfony/console/Symfony/Component/Console/Command/Command.php(257): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#12 /home/vagrant/project/vendor/laravel/framework/src/Illuminate/Console/Command.php(101): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#13 /home/vagrant/project/vendor/symfony/console/Symfony/Component/Console/Application.php(874): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#14 /home/vagrant/project/vendor/symfony/console/Symfony/Component/Console/Application.php(195): Symfony\Component\Console\Application->doRunCommand(Object(Illuminate\Queue\Console\WorkCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#15 /home/vagrant/project/vendor/symfony/console/Symfony/Component/Console/Application.php(126): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#16 /home/vagrant/project/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(94): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#17 /home/vagrant/project/artisan(36): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#18 {main}

  [Pheanstalk\Exception\ServerException]
  Job 13 NOT_FOUND: does not exist or is not reserved by client
```

**Comment by GrahamCampbell** [comment]
Not sure how we can fix this. It doesn't look broken to me?

**Comment by hskrasek** [comment]
I guess what I'm trying to find out is, is there anyway I can bury the job and not have the worker delete or release the job? Or will I just have to ignore this exception all together?

**Comment by hskrasek** [comment]
Or well, manually talk to beanstalkd instead of using what Laravel provides, but that would be a bummer :/

**Comment by GrahamCampbell** [comment]
Not sure. @taylorotwell probably knows more about this than me. :)

**Comment by hskrasek** [comment]
Help me @taylorotwell, you're my only hope :P. Thanks for chiming in though @GrahamCampbell

**Comment by mirzabusatlic** [comment]
+1

**Comment by GrahamCampbell** [comment]
Ping @taylorotwell.

**Comment by rulrok** [comment]
:+1:

**Comment by rulrok** [comment]
I believe I have figured that out.

On your `queue.php` config file, set the ttl for something larger than the default `60` value.

```php
		'beanstalkd' => [
			'driver' => 'beanstalkd',
			'host'   => env('BEANSTALKD_HOST'),
			'queue'  => 'default',
			'ttr'    => 600, //Example
		],
```

Do not forget to listen to your queue specifying a timeout equal

`php artisan queue:listen --timeout=600`

And in case you are executing some shell command using something like `Symfony\Component\Process` class, verify for timeout options when creating the object.

In my situation, I've specified it through the constructor

```php
 $this->process = new Process($s_COMMAND, null, null, null, 600); //600 seconds for timeout
 $this->process->run();
```
